[Telemetry] Add telemetry around the time it is taking for grabbing the telemetry stats

Bamieh opened the pull request:
# [IGNORE] CLOSED
Sorry about the noise please ignore this PR

Bamieh committed:
finish implementation

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

Bamieh committed:
use perf_hooks instea

Bamieh committed:
use perf_hooks instea

Bamieh committed:
finish code

Bamieh committed:
Merge branch 'main' of github.com:elastic/kibana into telemetry/collection_duration

CohenIdo committed:
[Cloud Posture] update CSP rules configuration template

jeramysoucy committed:
Resolves UI Breaks from Malformed Roles (#131915)

* Adds catch of exceptions from PrivilegeSerializer deserialize methods to role transform function.
Resolves 124808 where malformed Elasticsearch roles cause Kibana users and roles UIs to not display correctly.

* Adds logger to role transform functions

* File accidentally not saved prior to last commit

Kerry350 committed:
[Stack Monitoring] Fix error handling on pipelines page (#132055)

* Fix error handling on pipelines page

SergeyKleyman committed:
Fixed typo in Kibana's APM PHP Agent configuration page (#132122)

miltonhultgren committed:
[Stack Monitoring] Prevent exceptions in rule when no data present (#131332)

* [Stack Monitoring] Prevent exceptions in rule when no data present (#120111)

* Remove optional chaining

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
skip flaky jest suite (#131346)

Bamieh committed:
enable signing commit on new MBP

jughosta committed:
[Discover][Alerting] Prevent rule flyout from being open simultaneously with other popovers like search suggestions (#132108)

* [Discover][Alerting] Prevent rule flyout from being open simultaneously with other popovers like search suggestions

* [Discover][Alerting] Update tests

* [Discover][Alerting] Update tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mgiota committed:
[Actionable Observability] Application usage for Alerts, Cases, Rules views (#132006)

* add application usage to observability

* wrap Alerts page in a TrackApplicationView component

* define alerting pages in a config file

* add UsageCollectionSetup contract

* pass useCollection in the renderApp

* fix failing tests

* application usage for cases and rules

nreese committed:
[Maps] fix Map panels should not show the user controls in a dashboard report (#131970)

* [Maps] fix Map panels should not show the user controls in a dashboard report

* fix jest tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

parkiino committed:
[Security Solution][Admin][Policy] Fixes a bug where the last side nav item is hidden by the policy details sticky save bar (#131646)

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

stratoula committed:
Controls fixes the flakiness (#132107)

lcawl committed:
Minor UI text edits in Cases (#132074)

kqualters-elastic committed:
Make some data fetching hooks state updates atomic (#132069)

jbudz committed:
[artifacts testing] Re-enable FTR smoke tests (#132076)

* [artifacts testing] Re-enable FTR smoke tests

* --quiet

* docker disable chromium sandbox

* newline

dasansol92 committed:
[Security Solution][Endpoint] Adds generic functional test for artifac list pages (#131532)

* Adds generic functional test for artifac list pages

* Unify page objects to be generic. Moved test data into a mock file

* Fix wrong blocklist js comments

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

madirey committed:
[Security Solution] Better threshold rule error checking (#131088)

* Better threshold rule error checking

* Add more type dependent checks

* create/import/update/add-prepackaged

* Fix tests

* whoops

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

afharo commented on src/plugins/usage_collection/server/collector/collector_set.ts:
```diff
@@ -115,22 +112,40 @@ export class CollectorSet {
+    const timeoutMs = this.maximumWaitTimeForAllCollectorsInS * SECOND_IN_MS;
     const collectorsWithStatus: CollectorWithStatus[] = await Promise.all(
       [...collectors.values()].map(async (collector) => {
+        const wrappedPromise = performance.timerify(
```
I'm wondering: does it make sense to move both `performance.timerify` uses to a function that receives the fn to run and the name to assign the function?

IMO, it'll be easier to understand why we are doing the `Object.defineProperty`, which is essentially creating a named function with dynamic function names :)

WDYT?

Bamieh replied:
++

animehart committed:
[8.3] [Session View]Metadata Tab  (#131465)

* Metadata tab + new fields, edited Host tab jest, need to add more unit test and delete unused fields

* added accordion for Host, modified new fields

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* added more unit test for helper function, removed dropped fields from ECS spreadsheet, PR comments

* PR Comments, Fixing Checks issue, more Jest

* eslint

* changed the logic for checking which accordion not to render

* PR comments

* PR Comments, localized new accordion

* renaming file and component name from Host to Metadata

* merge main + PR comments

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

gchaps committed:
[DOCS] Fixes typo in Console doc (#132078)

* [DOCS] Fixes typo in Console doc

This PR fixes a typo in the Console doc.

* Update docs/dev-tools/console/console.asciidoc

* Update docs/dev-tools/console/console.asciidoc

ymao1 committed:
[Connectors] `ConnectorTokenClient` improvements (#131955)

* Throwing error if service now access token is null. Properly returning rejected promise

* Setting time to calculate token expiration to before the token is created

* Returning null access token if stored access token has invalid expiresAt date

* Adding response interceptor to delete connector token if using access token returns 4xx error

* Adding test for tokenRequestDate

* Handling 4xx errors in the response

* Fixing unit tests

* Fixing types

Commit:
[precommit-hook] autofix precommit check failures on CI (#132141)

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

ThomThomson committed:
[Dashboard][Controls] Make Text Field Based Options List Controls Case Insensitive (#131198)

* Adds case insensitive search and run past timeout option

semd committed:
[Security] New side navigation (#131437)

* initial implementation

* feature flag and setting added

* Simplify mobile view to use EuiCollapsibleNavGroup

* PR suggestions

* Fix mobile title

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* style with theme vars, tests

* more theming

* remove emoji

* test fixes

* fix users page cypress test

* snapshot updated

* new nav tests

* test fixes

* refactor css to styled components

* remove duplicated deeplink

* nav panel test added and conflig cleaning

Co-authored-by: cchaos <caroline.horn@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

yctercero committed:
[Security Solution][Exceptions] - Add back exceptions export success toast (#131952)

Addresses #88449

It appears that the success toaster code was deleted at some point as the text for the toaster was already there just not in use. Simple fix to add back in. Added check for toaster to existing cypress test.

Heenawter committed:
[Controls] [Dashboard] Allow existing controls to change type (#129385)

* Replace is half working

* Child embeddable listens for change in type

* Fix control order on replace

* Fix factory & remove duplicated onPanelAdded/Removed

* Comments + clean up code

* Set invalid editor state on type change

* Add functional tests and clean test code

* Comment out time slider tests

* Remove getReplacementPanelState

* Fix promise syntax

* Fix mutation

* Fix flaky test

* Fix conflicts

Commit:
[Unified observability] Show 10 items per page by default in the alerts table in the observability overview page. (#129633)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

jbudz committed:
[artifacts] Only upload beats artifacts if they exist (#132136)

* [artifacts] Only upload beats artifacts if they exist

Beats artifacts are downloaded as a side effect  of building the cloud
image.  At times (e.g. a staging build, where we don't have the correct
sha) we may skip the cloud image build and these artifacts will not be
available to upload.

* Update .buildkite/scripts/steps/artifacts/build.sh

Co-authored-by: Spencer <email@spalger.com>

Co-authored-by: Spencer <email@spalger.com>

lcawl committed:
[DOCS] Fixes data type for fields in find rules API (#132163)

lcawl committed:
[DOCS] Adds prerequisites for create, delete, get connector APIs (#131685)

opauloh committed:
[Session View] Close session viewer button fixes (#131910)

1Copenut committed:
Upgrade EUI to v55.1.2 (#131203)

* Bumping EUI to v55.1.0

* Updating package.json to latest EUI
* Adding one translation.

* Updating theme config to pass Jest Tests #4 / CaseView.

* Updating theme config to pass Jest Tests 3, CauseStacktrace.

* Updating six x-pack synthetic test snapshots.

* Updating snapshots for discover, reporting, security_solution.

* Replacing instances of spacerSizes with euiSize.

* Updating a number of snapshots for Emotion styles.

* Adding more snapshots for Emotion style upgrade.

* Updating seven snapshot tests for Emotion styles.

* Updating two snapshot tests for Emotion styles.

* Adding two more snapshots for License and Upload.

* Updating a Typescript check, and classname count in one unit test.

* Updating 1 snapshot and refining EuiLoadingChart selector to avoid off-by-one error.

* Bumping EUI to 55.1.1 for a change in EUI Flyout behavior.

* update newsfeed flyout to use shards

* snapshot update

* eui to v55.1.2

* update onClose

* onClose types

* reduce snapshot noise

* reduce snapshot noise

* Adding back Emotion output at request of Data Discovery reviewer.

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Greg Thompson <thompson.glowe@gmail.com>

Commit:
[ci] support skip:auto-commit label on CI (#132138)

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

mgiota committed:
[Actionable Observability] consume tags filter (#131220)

* consume tags filter

* add getRuleTagFilter to the observability start mock

* use loadRuleTags instead of loadRuleAggregations

* add an error when rule tag fail to load

* load tags errors in a useEffect

* fix failing unit tests

* load rule errors in a useEffect

* stop loading loadRuleTagsAggs twice

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Bamieh commented on src/plugins/usage_collection/server/collector/collector_stats/usage_collector_stats_collector.ts:
```diff
@@ -0,0 +1,94 @@
+    fetchExecutionDurationByType,
+  }: CollectorsStatsCollectorParams
+) => {
+  return usageCollection.makeUsageCollector<CollectorsStats>({
```
It's a very light-weight class. it is very lightweight. I can refactor it out however to avoid this altogether :+1:

afharo replied:
That's fine. If you think it won't cause any performance issues, I'm fine with keeping it this way :)

Bamieh commented on src/plugins/usage_collection/server/collector/collector_set.ts:
```diff
@@ -115,22 +112,40 @@ export class CollectorSet {
         });
 
-        return { isReadyWithTimeout, collector };
+        if (isReadyWithTimeout.timedout) {
+          return { isReadyWithTimeout, collector };
+        }
+
+        return {
+          isReadyWithTimeout: {
+            value: isReadyWithTimeout.value,
+            timedout: isReadyWithTimeout.timedout,
+          },
+          collector,
+        };
```
I believe i did that some time ago because it was the easiest way to inform the TS compiler that there are 2 different types here (timedout: true has no value while timedout false has a value). I will check if it will bs smart about it after the proposed change in the new TS version.

Bamieh commented on src/plugins/telemetry/schema/oss_plugins.json:
```diff
@@ -9610,6 +9610,124 @@
+        },
+        "fetch_duration_breakdown": {
+          "properties": {
+            "DYNAMIC_KEY": {
```
I think if we report it in an array we'd have to unwrap it to this object structure I am reporting to be able to visualize it and show top values. I can switch it to an array to reduce mapping explosion :+1:

afharo replied:
I had v3 indexer in mind: it's way easier to create the indexer just once that indexes each duration in separated documents and find the top values that way vs. revisiting the indexer configuration and the visualizations on every release just in case a new collector is added (and there is a new branch in the tree structure).

Commit:
[App Search] Add instructions for PDF Extraction to Crawler Overview (#132179)

renovate[bot] committed:
Update cypress (main) (#124070)

JordanSh committed:
used working example (#132201)

JordanSh committed:
added truncate option (#132202)

JordanSh committed:
Finding flyout changes (#132115)

academo committed:
[Cases] Show the Average time to close stat in the all cases page (#131909)

orouz committed:
[Cloud Posture] add pagination to resource findings table (#131422)

VladLasitsa committed:
[XY] Support `vis_dimension` type for accessors (#129612)

* added xy plugin.

* Added expressionXY limits.

* Added xy expression functions to the expression_xy plugin.

* Moved xy to a separate plugin.

* Small refactoring.

* Fixed types.

* Fixed import of scss.

* Fixed imports.

* Added required plugins.

* Fixed import

* Fixed types.

* Changed expression names.

* Fixed bugs, caused by the refactoring process.

* Fixed lens snapshots.

* Removed new line.

* Fixed xy_chart tests.

* Added lazy loading for xy chart.

* Fixed xy chart test.

* Fixed broken chart selectors.

* Fixed dashboard tests.

* dashboard test fixed.

* Fixed heatmap vis.

* Smokescreen test fixed.

* more fixes.

* async dashboard tests fixed.

* Fixed xy smokescreen tests selectors.

* fixed show_underlying_data tests.

* Updated snapshots.

* updated limits.

* Fixed more selectors

* Fixed persistent context test.

* Fixed some more test at ml.

* Fixed types and imports

* Fixed handlers.inspectorAdapters.tables.logDatatable

* Fixed logDatatable

* Translations fixed.

* Fixed "Visualize App ... cleans filters and query" test.

cleans filters and query

* Fixed "lens disable auto-apply tests" test.

* Updated dashboard tests.

* Fixed translations.

* Expression tests fixed.

* Cleaned up expression_xy.

* cleaned up lens xy_visualization.

* fixed more tests.

* Fix of tsvb.

* Fixed more tests.

* Fixed xy chart limits.

* Fixed new tests.

* Fixed types.

* Added extended layers expressions.

* Added support of tables at layers.

* Fixed tests.

* Fixed more tests.

* Fixed lens types.

* Added tables to layers.

* Checks fixed.

* updated tests.

* Fixed types.

* First try to fix merge conflicts.

* Fixed annotatations.

* Fixed types.

* Updated snapshots

* Fixed tests.

* Fixed dependencies.

* Fixed i18n.

* Moved XY state types to lens.

* Fixed more types.

* Update src/plugins/chart_expressions/expression_xy/README.md

Co-authored-by: Marta Bondyra <marta.bondyra@gmail.com>

* [CI] Auto-commit changed files from 'node scripts/build_plugin_list_docs'

* Removed yConfig from *Layers types

* Fixed styles.

* Fixed types.

* Removed not used utils and styles.

* Fixed types and tests.

* updated size.

* Added right behavior, related to the tables, comming from the expression.

* Fixed reference lines.

* Fixed jsdoc.

* Added annotations to layeredXyVIs.

* Fixed limits.

* Refactored the implementation to be reusable.

* Fixed undefined layers.

* Fixed empty arrays problems.

* Fixed input translations and removed not used arguments.

* Fixed missing required args error, and added required to arguments.

* Simplified expression configuration.

* Added strict to all the expressions.

* refactored code, according to the nit.

* Support visdimension type for accessors in data_layer and reference_line_layer

* Fix checks

* Moved dataLayer to the separate component.

* Fixed jest tests.

* Fixed tests.

* Some updates

* Refactored dataLayers helpers and xy_chart.

* More fixes of the expression

Added extendedYConfig for dataLayers.
Added yConfig for referenceLineLayers.
Fixed undefined id at tooltip.

* Fixed tests and snapshots.

* Icons at annotations and reference lines are strict.

* Fix CI

* axis extent validation added.

* Added checks to the legend config.

* fillOpacity usage validation is added.

* Fixed valueLabels argument options. Removed not used. Added validation for usage.

* Removed not used tests and imports.

* Fixed valueLabels and added migrations.

* Fixed type checks.

* Added test for the migrations.

* Fixed imports.

* Fixed types

* Fixed i18n checks.

# Conflicts:
#	src/plugins/chart_expressions/expression_xy/public/components/xy_chart.tsx

* Fixed imports and types.

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* Update src/plugins/chart_expressions/expression_xy/common/expression_functions/extended_annotation_layer.ts

Co-authored-by: Marta Bondyra <marta.bondyra@gmail.com>

* Removed extra extends.

* Update src/plugins/chart_expressions/expression_xy/common/expression_functions/annotation_layer.ts

Co-authored-by: Marta Bondyra <marta.bondyra@gmail.com>

* Added guard.

* Fixed the code duplication.

* Removed table from the annotation layer.

* Changed the `convertActiveDataFromIndexesToLayers` location.

* Added tests for convertActiveDataFromIndexesToLayers

* Reduced the bundle size a little bit.

* Reused strings and args.

* Refactored expression functions. Added asynchronous behavior.

* Fixed tests.

* Updated limits.

* Updated the limit.

* Fixed types.

* fixed types.

* Turned back layerIds.

* Removed convertActiveData from Lens.

* Added test to the layerIds generator.

* Fixed types.

* Fixed problems with resetting of the inspector.

* Fixed migrations.

* Removed types.

* Removed tones of `areFormatted` calculations.

* Fixed `isTimeViz` and `isHistogramViz` by replacing filteredLayers with dataLayers.

* Removed referenceLineLayers from the `groupAxesByType` fn.

* Added validation to the layeredXyVis.

* Fixed extent validation.

* Removed comments.

* Reduced limit.

* Added optimizations.

* Fixed floatingColumns error.

* Fixed types.

* Updated limits.

* Turned back extent validation.

* Fixed stacked error.

* Parallelized async import of functions.

* Decreased the complexity of the algorithm.

* Fixed snapshot.

* Some fixes

* Fix checks

* Fixed types

* Fix types

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* Added validation to the accessors.

* Validated accessors at the referenceLineLayer and extendedReferenceLineLayer.

* Fixed some style issues.

* Fixed imports.

* One more fix of import.

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* Move extended data layer fn in async file

* Some improvements

* Fix checks

* Fix validation

Co-authored-by: Yaroslav Kuznietsov <kuznetsov.yaroslav.yk@gmail.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Marta Bondyra <marta.bondyra@gmail.com>

Bamieh committed:
@afharo code review changes

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

Bamieh commented:
Sorry about the noise. rebased from my fork which was heavily outdated 😬  Closing. Will open a fresh PR with the relevant changes only
